enhancements/api-review/add-repositoryMirrors-spec.md

```diff
@@ -0,0 +1,175 @@
+	// tag can potentially yield different images, depending on which endpoint
+	// we pull from.  Forcing digest-pulls for mirrors avoids that issue.
+	MirrorByDigestOnly bool `toml:"mirror-by-digest-only,omitempty"`
+}
+```
+
+The ImageContentSourcePolicy CRD has the cluster-wide information about handle the mirrors only when pulling the images that are referenced by their digests.
+It will now watch for mirrors configured through spec `repositoryMirrors` to not require digest reference.
+
+An example ImageContentSourcePolicy file will look like:
+```yaml
+apiVersion: operator.openshift.io/v1alpha1
+kind: ImageContentSourcePolicy
+metadata:
+  name: ubi8repo
+spec:
+  repositoryMirrors:
+  - mirrors:
+    - example.io/example/ubi-minimal 
+    source: registry.access.redhat.com/ubi8/ubi-minimal
+```
+
+### User Stories
+
+#### As a user, I would like to pull images from mirror using tag, without digests reference
+The user need to define multiple ImageContentSourcePolicies, used by apps/manifests that don't use digests to pull
+the images when working with disconnected environments or pulling from registries that act as transparent pull-through proxy cache.
+The user can pull image without digest by configuring `repositoryMirrors` spec in the ImageContentSourcePolicy file.
+And create the ImageContentSourcePolicy project. Once this is done, the images can be pulled from the mirrors without the digest referenced.
+
+### Implementation Details/Notes/Constraints [optional]
+
+Implementing this enhancement requires changes in:
+- openshift/api
+- openshift/machine-config-operator
+
+This is an example of the ImageContentSourcePolicy file:
+
+```yaml
+apiVersion: operator.openshift.io/v1alpha1
+kind: ImageContentSourcePolicy
+metadata:
+  name: ubi8repo
+spec:
+  repositoryMirrors:
+  - mirrors:
+    - example.io/example/ubi-minimal 
+    source: registry.access.redhat.com/ubi8/ubi-minimal 
+```
+
+The ImageContentSourcePolicy file will create a drop-in file at `/etc/containers/registries.conf` that looks like:
+
+```toml
+unqualified-search-registries = ["registry.access.redhat.com", "docker.io"]
+[[registry]]
+  location = "registry.access.redhat.com/ubi8/"
+  insecure = false
+  blocked = false
+  # No mirror-by-digest-only = true configured
+  prefix = ""
+
+  [[registry.mirror]]
+    location = "example.io/example/ubi8-minimal"
+    insecure = false
+```
+
+If both `repositoryMirrors` and `repositoryDigestMirrors` are configured for same “source” repository, the ImageContentSourcePolicy will be rejected.
+
+### Risks and Mitigations
+
+To pull images from mirror registries without the digest specifications, this could lead to returning different image version from different registry if the image tag mapping is out of sync. This does not bring risks for ICSP usage since it does not influence the openshift deployments, which required image using digests to avoid different versions.
```

Comment:
I think the "This does not bring risks..." sentence is a bit off base.  It would be more clear to say the OpenShift core won't consume this feature at all, so it is not exposed to the risks that anyone who actually uses the feature will be exposed to.